docs/guide/ingress/ingress_class.md

```diff
@@ -0,0 +1,140 @@
+    kind: IngressClassParams
+    metadata:
+      name: awesome-class
+    spec:
+      group:
+        name: my-group
+    ```
+
+### IngressClassParams specification
+
+#### spec.namespaceSelector
+`namespaceSelector` is an optional setting, It follows standard label selector semantics.
+
+Cluster administrators can use `namespaceSelector` field to restrict the namespaces of Ingresses that are allowed to specify the IngressClass.
+
+1. If `namespaceSelector` specified, only Ingresses in selected namespaces can use IngressClasses with this parameter. The controller will refuse to reconcile for Ingresses that violates `namespaceSelector`.
+2. If `namespaceSelector` un-specified, all Ingresses in any namespace can use IngressClasses with this parameter.
+
+#### spec.group
+
+`group` is an optional setting.  The only available sub-field is `group.name`.
+
+Cluster administrators can use `group.name` field to denote the groupName for all Ingresses belong to this IngressClass.
+
+1. If `group.name` specified, all Ingresses with this IngressClass will belong to the same IngressGroup specified and result in a single ALB.
+2. If `group.name` un-specified, Ingresses with this IngressClass can continue to use `alb.ingress.kubernetes.io/group.name` annotation to specify IngressGroup. Ingresses belong to same IngressClass can form different IngressGroups via annotation.
+
+#### spec.scheme
+
+`scheme` is an optional setting. The available options are `internet-facing` or `internal`.
+
+Cluster administrators can use `scheme` field to restrict the scheme for all Ingresses belong to this IngressClass.
+
+1. If `scheme` specified, all Ingresses with this IngressClass will have the specified scheme.
+2. If `scheme` un-specified, Ingresses with this IngressClass can continue to use `alb.ingress.kubernetes.io/scheme annotation` to specify scheme.
+
+#### spec.ipAddressType
+
+`ipAddressType` is an optional setting. The available options are `ipv4` or `dualstack`.
+
+Cluster administrators can use `ipAddressType` field to restrict the ipAddressType for all Ingresses belong to this IngressClass.
+
+1. If `ipAddressType` specified, all Ingresses with this IngressClass will have the specified ipAddressType.
+2. If `ipAddressType` un-specified, Ingresses with this IngressClass can continue to use `alb.ingress.kubernetes.io/ip-address-type` annotation to specify ipAddressType.
+
+#### spec.tags
+
+`tags` is an optional setting.
+
+Cluster administrators can use `tags` field to specify the custom tags for AWS resources provisioned for all Ingresses belong to this IngressClass.
+
+1. If `tags` specified, AWS resources provisioned for all Ingresses with this IngressClass will have the specified tags.
+2. Customers can also use controller-level flag `--default-tags`  or `alb.ingress.kubernetes.io/tags` annotation to specify custom tags. These tags will be merged together based on tag-key. If same tag-key appears in multiple sources, the priority is as follows:
```

Comment:
```suggestion
2. You can also use controller-level flag `--default-tags`  or `alb.ingress.kubernetes.io/tags` annotation to specify custom tags. These tags will be merged together based on tag-key. If same tag-key appears in multiple sources, the priority is as follows:
```
(based on [address the reader as you](https://kubernetes.io/docs/contribute/style/style-guide/#address-the-reader-as-you) from the Kubernetes SIG Docs style guide).